assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -393,6 +495,32 @@ export const resolvers = {
 };
 
 export const selectors = {
+	/**
+	 * Gets all seen notifications view date arrays.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object} Object with notification IDs as keys and view count as values.
+	 */
+	getSeenNotifications( state ) {
+		return state.seenNotifications;
+	},
+
+	/**
+	 * Gets the dates when a specific notification was seen.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state          Data store's state.
+	 * @param {string} notificationID Notification ID.
+	 * @return {Array} Array of dates when the notification was seen.
+	 */
+	getNotificationSeenDates( state, notificationID ) {
+		const { seenNotifications } = state;
+		return seenNotifications[ notificationID ] || [];
```

Comment:
Arguably this is a bit more intentional about what we're looking for here 😅 

```suggestion
		return seenNotifications[ notificationID ] ?? [];
```